tests/schedulers/test_lsf.py

```diff
@@ -32,162 +32,206 @@
+    assert num_machines == parsed_num_machines
+    assert allocated_machines == parsed_allocated_machines
+
+    assert [j.requested_wallclock_time_seconds for j in job_list if j.job_id == '764254593'][0] == 60  # pylint: disable=invalid-name
+    assert [j.wallclock_time_seconds for j in job_list if j.job_id == '764255172'][0] == 9
+    assert [j.wallclock_time_seconds for j in job_list if j.job_id == '764245175'][0] == 4785
+    current_year = datetime.datetime.now().year
+    assert [j.submission_time for j in job_list if j.job_id == '764245175'
+            ][0] == datetime.datetime(current_year, 12, 31, 23, 40)
+
+    # Important to enable again logs!
+    logging.disable(logging.NOTSET)
+
+
+def test_submit_script():
+    """Test the creation of a simple submission script"""
+    from aiida.schedulers.datastructures import JobTemplate
+    from aiida.common.datastructures import CodeInfo, CodeRunMode
+
+    scheduler = LsfScheduler()
+
+    job_tmpl = JobTemplate()
+    job_tmpl.shebang = '#!/bin/bash'
+    job_tmpl.uuid = str(uuid.uuid4())
+    job_tmpl.job_resource = scheduler.create_job_resource(tot_num_mpiprocs=2, parallel_env='b681e480bd.cern.ch')
+    job_tmpl.max_wallclock_seconds = 24 * 3600
+    code_info = CodeInfo()
+    code_info.cmdline_params = ['mpirun', '-np', '2', 'pw.x', '-npool', '1']
+    code_info.stdin_name = 'aiida.in'
+    job_tmpl.codes_info = [code_info]
+    job_tmpl.codes_run_mode = CodeRunMode.SERIAL
+    job_tmpl.account = 'account_id'
+
+    submit_script_text = scheduler.get_submit_script(job_tmpl)
+
+    assert submit_script_text.startswith('#!/bin/bash')
+
+    assert '#BSUB -rn' in submit_script_text
+    assert '#BSUB -n 2' in submit_script_text
+    assert '#BSUB -W 24:00' in submit_script_text
+    assert '#BSUB -G account_id' in submit_script_text
+    assert "'mpirun' '-np' '2' 'pw.x' '-npool' '1' < 'aiida.in'" in submit_script_text
+
+
+def test_create_job_resource():
+    """
+    Test to verify that script fails if we specify only num_machines or wrong tot_num_mpiprocs
+    """
+    from aiida.schedulers.datastructures import JobTemplate
 
-        retval = 0
-        stdout = BJOBS_STDOUT_TO_TEST
-        stderr = ''
-        job_list = scheduler._parse_joblist_output(retval, stdout, stderr)
+    scheduler = LsfScheduler()
+    job_tmpl = JobTemplate()
+
+    with pytest.raises(TypeError):
+        job_tmpl.job_resource = scheduler.create_job_resource(tot_num_mpiprocs='Not-a-Number')
+
+    with pytest.raises(TypeError):
+        job_tmpl.job_resource = scheduler.create_job_resource(
+            num_machines=1,
+            num_mpiprocs_per_machine=1,
+        )
+
+    class StrCastRaiser:
+
+        def __str__(self):
+            raise TypeError('This type cannot be casted to str')
+
+    with pytest.raises(TypeError):
+        job_tmpl.job_resource = scheduler.create_job_resource(
+            tot_num_mpiprocs=2,
+            parallel_env=StrCastRaiser(),
+        )
+
```

Comment:
I think this is my [hackest hack](https://www.youtube.com/watch?v=uOa-ObWPAKg) so far...